cadence/contracts/FindRegistry.cdc

```diff
@@ -0,0 +1,205 @@
+    pub event Cleared(registry: UInt64, id: UInt64)
+    pub event UpdatedTTL(registry: UInt64, ttl: UInt64)
+    pub event Deleted(registry: UInt64, owner: Address?)
+
+    /**//////////////////////////////////////////////////////////////
+    //                         INTERFACES                          //
+    /////////////////////////////////////////////////////////////**/
+
+    pub resource interface RegistryPublic {
+        pub fun get(id: UInt64): RegistryEntry?
+        pub fun contains(id: UInt64, account: Address): Bool
+        pub fun clearExpired(ids: [UInt64])
+    }
+
+    pub resource interface RegistryPrivate {
+        pub fun add(id: UInt64, account: Address)
+        pub fun remove(id: UInt64, account: Address)
+        pub fun updateTTL(ttl: UInt64)
+    }
+
+    /**//////////////////////////////////////////////////////////////
+    //                         RESOURCES                           //
+    /////////////////////////////////////////////////////////////**/
+
+    /// Registry
+    /// - A resource that represents a registry
+    /// - contains a dictionary of UUIDs to registry entries and a TTL
+    pub resource Registry: RegistryPublic, RegistryPrivate {
+        /// A dictionary of UUIDs to registry entries
+        pub var registry: {UInt64: RegistryEntry}
+
+        /// The registry's TTL
+        pub var registryTTL: UInt64
+
+        /// add
+        /// Add an account to the registry for a UUID
+        /// @params id: The UUID to add to the registry
+        /// @params account: The account to add to the registry
+        pub fun add(id: UInt64, account: Address) {
+            let blockHeight: UInt64 = getCurrentBlock().height
+
+            // check if registy exists for this UUID
+            if self.registry[id] == nil {
+                // create a new registry entry
+                let registryEntry = RegistryEntry(blockHeight: blockHeight)
+                registryEntry.addAccount(account: account)
+                self.registry[id] = registryEntry
+            } else {
+                // get the registry entry
+                let registryEntry = self.registry[id]!
+
+                // check if the block height is expired
+                if registryEntry.blockHeight + self.registryTTL < blockHeight {
+                    registryEntry.addAccount(account: account)
+                    // emit event
+                    emit Added(registry: self.uuid, address: account, id: id, blockHeight: blockHeight)
+                } else {
+                   return
+                }
+            }
+        }
+
+        /// remove
+        /// Remove an account from the registry for a UUID
+        /// @params id: The UUID to remove from the registry
+        /// @params account: The account to remove from the registry
+        pub fun remove(id: UInt64, account: Address) {
+            // check if registy exists for this UUID
+            if self.registry[id] != nil {
```

Comment:
nit: i prefer return home early, so reverting the if and returning instead of nesting.